backend/views/views.py

```diff
@@ -34,12 +36,43 @@ def admin():
     return render_template('admin_page.html')
 
 
-@app.route('/userpage')
+@app.route('/userpage', methods=['GET'])
 @admin_permissions
 def user_page():
+    form = SearchForm(request.form)
     users = db.session.query(User, Role).filter(
         User.role_id == Role.id).order_by(User.id).all()
-    return render_template('user_page.html', users=users)
+    if request.args.get('search') != "" and request.args.get('search') >= MIN_SEARCH_STR:
```

Comment:
Please use `form.validate()` method. In that case you will check search `search` field to min length.

Reply:
This method does not work too, maybe problem in form. I don`t know

Reply:
The root of cause of the issue is that you create WTForm object using `request.form`, but this variable is only available for POST requests, and for GET ones you should use `request.args`. So init statement should be like:
```
form = SearchForm(request.args)
```

Reply:
I found an error "CSRF token is missing". When in config.py "WTF_CSRF_ENABLED = False"
method form.validate() is work

Reply:
add {{ form.hidden_tag() }} into your form

Reply:
There is no sense to protect search forms from CSRF attacks, you may [disable CSRF](https://flask-wtf.readthedocs.io/en/stable/form.html#module-flask_wtf) protection for `SearchForm` by providing `csrf_enabled=False` parameter:
```
form = SearchForm(request.args, csrf_enabled=False)
```

Reply:
Finally it works :) thanks